### DISCUSSION TITLE
Confusion regarding async calling the "sync" lambda invoke

### DISCUSSION THREAD

**Discussion by ghost** [aio-libs/aiobotocore#1009] [Q&A] [answered]
Hello,

While this is a aioboto3 question I reckon this is more aiobotocore related since I'm called the lambda client. If it is the wrong place please say so and i'll remove it.

Take for example the following code:

```
session = aioboto3.Session()

async def call_invoke(argument, lambda_client):
    response = await lambda_client.invoke(
        InvocationType="RequestResponse",
        FunctionName="to_be_invoked_lambda",
        Payload=json.dumps(("test", argument)))
    return json.loads(await response['Payload'].read())


async def example(list_of_arguments):
    async with session.client('lambda') as lambda_client:
        task_list = []
        for arg in list_of_arguments:
            task_list.append(call_invoke(arg, lambda_client))
        return await asyncio.gather(*task_list)


def run_invokes(list_of_arguments: list) -> list:
    return asyncio.run(run_process(list_of_arguments))
```

The code runs fine. For every argument in list_of_arguments an invoke is made.
Though what I've noticed is that it's limited to 10 concurrent calls regardless of the concurrency value of lambda to_be_invoked_lambda
to_be_invoked_lambda only takes 7 ms on average anyway.
Ie when I look with xray I never see more then 10 invokes of to_be_invoked_lambda with initialization.

I've already tried seperating the invoke call and the read call in seperate async functions but that doesn't really change anything.

Is there some hardcoded limit in aiobotocore that limits async calls to the lambda invoke when it's "RequestResponse"?

**Comment by thehesiod** [comment, marked as answer]
ya, by default both botocore and aiobotocore has a 10 connection pool limit. you can increase this as needed via the Config/AioConfig object

**Reply by ghost** [reply]
Thanks for the assistance. This was obvious in hindsight.
Sadly it seems there is quite a bit of overhead in setting up more connections.

Upping the limit to 20 only reduces the total time by 10~20% and upping it to 100 even less.

I'm starting to think that 10 limit might be there for a reason.
